src/runtime-rs/crates/resource/src/rootfs/nydus_rootfs.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SP&DX-License-Identifier: Apache-2.0
+//
+
+use super::{Rootfs, TYPE_OVERLAY_FS};
+use crate::{
+    rootfs::{HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR, ROOTFS},
+    share_fs::{
+        do_get_guest_path, do_get_guest_virtiofs_path, get_host_rw_shared_path, rafs_mount,
+        PASSTHROUGH_FS_DIR,
+    },
+};
+use agent::Storage;
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use hypervisor::Hypervisor;
+use kata_sys_util::mount;
+use kata_types::mount::{Mount, NydusExtraOption};
+use std::fs;
+
+// Used for nydus rootfs
+pub(crate) const NYDUS_ROOTFS_TYPE: &str = "fuse.nydus-overlayfs";
+const NYDUS_ROOTFS_V5: &str = "v5";
+// Used for Nydus v6 rootfs version
+const NYDUS_ROOTFS_V6: &str = "v6";
+
+const SNAPSHOTDIR: &str = "snapshotdir";
+const KATAOVERLAYDEVTYPE: &str = "overlayfs";
+pub(crate) struct NydusRootfs {
+    guest_path: String,
+    rootfs: Storage,
+}
+
+impl NydusRootfs {
+    pub async fn new(h: &dyn Hypervisor, sid: &str, cid: &str, rootfs: &Mount) -> Result<Self> {
+        let extra_options: NydusExtraOption =
+            get_nydus_extra_options(rootfs).context("failed to parse nydus extra options")?;
+        info!(sl!(), "extra_option {:?}", &extra_options);
+        let rafs_meta = &extra_options.source;
+        let (rootfs_storage, rootfs_guest_path) = match extra_options.fs_version.as_str() {
+            NYDUS_ROOTFS_V5 => {
+                let rafs_mnt =
+                    do_get_guest_virtiofs_path(HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR, cid, true);
```

Comment:
I think it's better to rename to `do_get_guest_path()` or `do_get_guest_share_path()`.

Reply:
fixed